From 601ebfacee324b1082ddf6d4df13ce791ab5cbe1 Mon Sep 17 00:00:00 2001
From: Sang Ngo <sang.ngo.yf@renesas.com>
Date: Tue, 3 Feb 2026 14:17:42 +0700
Subject: [PATCH] Update MIN MAX definination to compatible with cip kernel

Since commit 1a251f52cfdc417c84411a056bc142cbd77baef4, MIN MAX
is already defined in the cip kernel. So, it should no longer be defined
in Mali driver

Signed-off-by: Sang Ngo <sang.ngo.yf@renesas.com>
---
 drivers/gpu/arm/midgard/mali_malisw.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/arm/midgard/mali_malisw.h b/drivers/gpu/arm/midgard/mali_malisw.h
index ce0c76b..c684052 100644
--- a/drivers/gpu/arm/midgard/mali_malisw.h
+++ b/drivers/gpu/arm/midgard/mali_malisw.h
@@ -27,8 +27,9 @@
 #define _MALISW_H_
 
 #include <linux/version.h>
+#include <linux/minmax.h>
 
-#if (KERNEL_VERSION(6, 11, 0) > LINUX_VERSION_CODE)
+#ifndef MIN
 /**
  * MIN - Return the lesser of two values.
  * @x: value1
@@ -38,7 +39,8 @@
  * Refer to MAX macro for more details
  */
 #define MIN(x, y) ((x) < (y) ? (x) : (y))
-
+#endif
+#ifndef MAX
 /**
  * MAX - Return the greater of two values.
  * @x: value1
-- 
2.25.1

